x-pack/plugins/ml/public/application/components/import_export_jobs/import_jobs_flyout/import_jobs_flyout.tsx

```diff
@@ -0,0 +1,493 @@
+      jobsExist(existsChecks, true).then((resp) => {
+        jobIds.forEach((j) => {
+          if (j.valid === true) {
+            // only apply the exists result if the previous checks are ok
+            const hh = resp[j.id];
+            j.valid = !hh.exists;
+            j.invalidMessage = hh.exists ? jobExists : '';
+          }
+        });
+        setJobIds([...jobIds]);
+        setValidatingJobs(false);
+      });
+    }
+  }, [idsMash, jobIds]);
+
+  useDebounce(validateIds, 400, [idsMash]);
+
+  const renameJob = useCallback(
+    (id: string, i: number) => {
+      jobIds[i].id = id;
+      jobIds[i].valid = false;
+      setJobIds([...jobIds]);
+      setIdsMash(jobIds.map(({ id: jId }) => jId).join(''));
+      setValidatingJobs(true);
+    },
+    [jobIds]
+  );
+
+  const DeleteJobButton: FC<{ index: number }> = ({ index }) => (
+    <EuiButtonIcon
+      iconType="trash"
+      aria-label="Delete"
+      color={deleteDisabled ? 'subdued' : 'danger'}
+      disabled={deleteDisabled}
+      onClick={() => deleteJob(index)}
+    />
+  );
+
+  return (
+    <>
+      <FlyoutButton onClick={toggleFlyout} isDisabled={isDisabled} />
+
+      {showFlyout === true && isDisabled === false && (
+        <EuiFlyout onClose={() => setShowFlyout(false)} hideCloseButton size="m">
+          <EuiFlyoutHeader hasBorder>
+            <EuiTitle size="m">
+              <h2>
+                <FormattedMessage
+                  id="xpack.ml.importExport.importFlyout.flyoutHeader"
+                  defaultMessage="Import jobs"
+                />
+              </h2>
+            </EuiTitle>
+          </EuiFlyoutHeader>
+          <EuiFlyoutBody>
+            <>
+              <div style={{ textAlign: 'center' }}>
+                <EuiFilePicker
+                  disabled={importing}
+                  fullWidth={true}
+                  id="filePicker"
+                  initialPromptText={i18n.translate(
+                    'xpack.ml.importExport.importFlyout.fileSelect',
+                    {
+                      defaultMessage: 'Select or drag and drop a file',
+                    }
+                  )}
+                  onChange={(files) => onFilePickerChange(files)}
+                  className="file-datavisualizer-file-picker"
+                />
+              </div>
+
+              {showFileReadError && <CannotReadFileCallout />}
+
+              {totalJobsRead > 0 && jobType !== null && (
+                <>
+                  <EuiSpacer size="l" />
+                  {jobType === 'anomaly-detector' && (
+                    <FormattedMessage
+                      id="xpack.ml.importExport.importFlyout.selectedFiles.ad"
+                      defaultMessage="{num} anomaly detection {num, plural, one {job} other {jobs}} read from file"
+                      values={{ num: totalJobsRead }}
+                    />
+                  )}
+
+                  {jobType === 'data-frame-analytics' && (
```

Comment:
`JobType` is a string literal type of either `anomaly-detector` or `data-frame-analytics`.
It's used throughout our code, originally added for job space awareness.
https://github.com/elastic/kibana/blob/59b56e5ac107c24ad69f34f6ac9ad8972ae82a27/x-pack/plugins/ml/common/types/saved_objects.ts#L8 

We have discussed it before on the team, but personally I prefer to rely on typescript to control the allowed values for it rather than have to import a constant everywhere it is used.